R/data.table.R

```diff
@@ -2518,7 +2517,7 @@ setnames <- function(x,old,new,skip_absent=FALSE) {
   # But also more convenient than names(DT)[i]="newname"  because we can also do setnames(DT,"oldname","newname")
   # without an onerous match() ourselves. old can be positions, too, but we encourage by name for robustness.
   if (!is.data.frame(x)) stop("x is not a data.table or data.frame")
-  if (length(names(x)) != length(x)) stop("dt is length ",length(dt)," but its names are length ",length(names(x)))
+  if (length(names(x)) != length(x)) stop("dt is length ",length(x)," but its names are length ",length(names(x)))
```

Comment:
this is #3194

Reply:
`dt is length` could be `x argument is length`.
`dt` is not present in context of that call